pkg/scheduler/framework/plugins/noderesources/resource_allocation.go

```diff
@@ -99,7 +99,7 @@ func calculateResourceAllocatableRequest(nodeInfo *framework.NodeInfo, pod *v1.P
 	case v1.ResourceEphemeralStorage:
 		return nodeInfo.Allocatable.EphemeralStorage, (nodeInfo.Requested.EphemeralStorage + podRequest)
 	default:
-		if schedutil.IsScalarResourceName(resource) {
+		if _, exists := nodeInfo.Allocatable.ScalarResources[resource]; exists {
```

Comment:
This is functional okay, only thing that might worth discussing is that  if a pod requests a scalar resource that doesn't exist  on that host, 

originally, it might return "0" and request quota from that pod.
now, it will be handled as an unknown resource, write the log and return "0", "0".

Reply:
Thanks for bringing this up. Yes, the log message will be printed even for scalar resources. Luckily, `requested resource %v not considered for node score calculation` log message is still meaningful even for the case when the scalar source is not found on a node.

Reply:
Another way is to drop the condition completely since `nodeInfo.Allocatable.ScalarResources[resource]` will given `0` when `resource` key does not exist. Returning non-zero `podRequest` if there's any.

Reply:
+1 on the drop of the condition, lets hold and see what's reviewers thinking about this.

Reply:
Lets keep the check because otherwise you will end up returning an above zero node requested resources for a node that has zero of this resource (line 103 below).

Reply:
right, but if you don't check, then you will be returning `(nodeInfo.Requested.ScalarResources[resource] + podRequest)`, so it will be `podRequest` not 0.

Reply:
original code will return `podRequest` instead of 0 when the scalar resource doesn't available on that host.
either `podRequest` or 0 is acceptable by far since it's not consumed anywhere when the allocatable resource is 0.